gems/sorbet-runtime/test/types/virally_install_hooks.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+class Opus::Types::Test::VirallyInstallHooksTest < Critic::Unit::UnitTest
+  it "does not give the hooks when including a non-viral module" do
+    m = Module.new do
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      include m
+    end
+    assert_equal(false, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+
+  it "does not give the hooks when extending a non-viral module" do
+    m = Module.new do
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      extend m
+    end
+    assert_equal(false, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+
+  it "gives the hooks when including a viral module" do
+    m = Module.new do
+      T::Private::Methods.instance_variable_get(:@virally_install_hooks) << self
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      include m
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+
+  it "gives the hooks when extending a viral module" do
+    m = Module.new do
+      T::Private::Methods.instance_variable_get(:@virally_install_hooks) << self
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      extend m
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+end
```

Comment:
It seems like the primary property we're trying to test is not the one being tested here. Our use of `instance_variable_get` makes me think that this test is really just testing that instance_variable_set / instance_variable_get still work in Ruby. The thing we really want to know is if we do this:

```
include m1
include m2
```

that our method added hooks get run for any methods defined in `m2`, right? Instead of reaching into instance variables, can we test that property directly?

Reply:
Addressed above.